GraphColoring/src/coroutine_generator.h

```diff
@@ -0,0 +1,174 @@
+#pragma once
+
+#include <coroutine>
+#include <exception>
+#include <iostream>
+#include <ranges>
+#include <tuple>
+#include <utility>
+
+
+template <typename promise_type>
```

Comment:
Looks like this is copied from https://codereview.stackexchange.com/questions/250918/experimenting-with-c20-coroutines-to-create-python-style-generator-functions
What for?

Reply:
Yes, I have taken this article as one of the examples for C++20 coroutuines. I was using boost coroutines but the solution with boost ended up too verbose and complex so I decided to try C++20 coroutines. As far as I understood std::generator was supposed to get into standard but did not make it. I think it is very useful and, hence, I used this implementation from the article.

Reply:
Also I didnt want to use cppcoro since its installation is not straight-forward to add here

Reply:
I guess to use cppcoro would be the best solution actually, so I am going to integrate it instead